### ISSUE TITLE
[question]: no timeout waiting for slug?

### ISSUE THREAD

**Issue by aledbf** [deis/builder#118]
during the build an error occurred (not important):
````
       npm ERR! 404
       npm ERR! 404 Note that you can also install from a
       npm ERR! 404 tarball, folder, http url, or git url.

       npm ERR! Please include the following file with any support request:
       npm ERR!     /tmp/build/npm-debug.log

-----> Build failed

       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys

       If you're stuck, please submit a ticket so we can help:
       https://help.heroku.com/

       Love,
       Heroku

remote: mc: <ERROR> Unable to stat ‘http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push’. The specified key does not exist.
running [mc -C /var/minio-conf --quiet ls http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push]
remote: mc: <ERROR> Unable to stat ‘http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push’. The specified key does not exist.
running [mc -C /var/minio-conf --quiet ls http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push]
remote: mc: <ERROR> Unable to stat ‘http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push’. The specified key does not exist.
running [mc -C /var/minio-conf --quiet ls http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push]
remote: mc: <ERROR> Unable to stat ‘http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push’. The specified key does not exist.
running [mc -C /var/minio-conf --quiet ls http://10.3.3.23:9000/git/home/srv-f22:git-6e3bc01dd7494525d5502900260fc4dc908bd0a8/push]

(this command continues "forever")
```

In [here](https://github.com/deis/builder/blob/master/pkg/gitreceive/build.go#L252) or #80 waits forever for the file. 
Maybe it should throw an error after a while?

**Comment by arschles** [comment]
@aledbf yes, absolutely. I'm planning to add a timeout in #106 (which will close #80) to time out after a configurable amount of time (with a sensible default).

Additionally, it's worth noting that #115 will programmatically watch the Kubernetes event stream for the builder pod, so I may be able to remove this polling mechanism altogether in that PR.

**Comment by aledbf** [comment]
@arschles awesome

**Comment by aledbf** [comment]
Closed by #133
